framework/src/org/checkerframework/framework/type/BoundsInitializer.java

```diff
@@ -211,10 +211,13 @@ public Void visitUnion(AnnotatedUnionType type, Void aVoid) {
 
         @Override
         public Void visitArray(AnnotatedArrayType type, Void aVoid) {
-            final BoundPathNode componentNode = addPathNode(new ArrayComponentNode());
-            type.setComponentType(replaceOrVisit(type.getComponentType()));
-            removePathNode(componentNode);
-
+            if (!TypesUtils.isPrimitive(type.getComponentType().getUnderlyingType())) {
```

Comment:
It's impossible to tell from the class description whether this is the right behavior or not.
Is the intent to add annotations to all types in the bounds? Then why wouldn't you want to apply defaults to primitives that appear as array components?
If you understand the purpose of this class, could you expand on its javadoc?